frontend/src/components/App/Layout.tsx

```diff
@@ -44,11 +54,133 @@ const useStyle = makeStyles(theme => ({
 
 export interface LayoutProps {}
 
+const CLUSTER_FETCH_INTERVAL = 10 * 1000; // ms
+
+interface Config {
+  [prop: string]: any;
+}
+
+function ClusterNotFoundPopup(props: {
+  isClusterNotAvailable: boolean;
+  changeClusterNotAvailableStateHandler: (isClusterNotAvailable: boolean) => void;
+}) {
+  const { isClusterNotAvailable, changeClusterNotAvailableStateHandler } = props;
+  const theme = useTheme();
+  const cluster = getCluster();
+  const { t } = useTranslation('frequent');
+  const isSidebarOpen = useTypedSelector(state => state.ui.sidebar.isSidebarOpen);
+  const history = useHistory();
+
+  if (!cluster) {
+    return null;
+  }
+
+  if (!isClusterNotAvailable) {
+    return null;
+  }
+  return (
+    <>
+      <Box
+        display={'flex'}
+        justifyContent="center"
+        style={{
+          position: 'absolute',
+          color: theme.palette.common.white,
+          textAlign: 'center',
+        }}
+        bgcolor={'error.main'}
+        zIndex={1400}
+        width={'100%'}
+        p={0.5}
+        alignItems="center"
+      >
+        <Box ml={isSidebarOpen ? '-25vw' : 0}>
+          {t(`Something went wrong with cluster ${cluster}`)}
+        </Box>
+        <Box
+          style={{ cursor: 'pointer' }}
+          color={'error.main'}
+          bgcolor={theme.palette.common.white}
+          ml={1}
+          p={0.5}
+          onClick={() => {
+            history.replace('/');
+            changeClusterNotAvailableStateHandler(false);
+          }}
+        >
+          {t('frequent| Take me to cluster chooser')}
+        </Box>
+      </Box>
+    </>
+  );
+}
+
 export default function Layout({}: LayoutProps) {
   const classes = useStyle();
   const arePluginsLoaded = useTypedSelector(state => state.ui.pluginsLoaded);
-
+  const dispatch = useDispatch();
+  const clusters = useTypedSelector(state => state.config.clusters);
   const { t } = useTranslation('frequent');
+  const [isClusterNotAvailable, setIsClusterNotAvailable] = useState(false);
+
+  useEffect(() => {
```

Comment:
Probably all of this new logic should go into it's own component. New components should go into their own file named after the component. In this case it would make sense if all the related parts are in components inside a folder of components, and then only one component is imported and used. So that all the pieces implementing this feature are together in one place.

Reply:
umm i am not sure if we want to create a new component out of this, the new addition just add a new fetchConfig function and passes new set of state to the already present RouteSwitcher component